pkg/downloader/chart_downloader.go

```diff
@@ -97,7 +97,16 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 
 	name := filepath.Base(u.Path)
 	destfile := filepath.Join(dest, name)
-	if err := ioutil.WriteFile(destfile, data.Bytes(), 0644); err != nil {
+	tmpfile, err := ioutil.TempFile(dest, "tmp")
+	if err != nil {
+		return "", nil, err
+	}
+	tmpfilename := tmpfile.Name()
+	if err := ioutil.WriteFile(tmpfilename, data.Bytes(), 0644); err != nil {
```

Comment:
I believe that the `0644` permissions will only be applied if `ioutil.WriteFile` creates a file which does not already exist. Since we create `tmpfile` with `TempFile`, I believe that we need to use `os.Chmod` to change the permissions of the `TempFile`.